common/utils/src/main/scala/org/apache/spark/internal/Logging.scala

```diff
@@ -391,7 +392,8 @@ trait Logging {
 private[spark] object Logging {
   @volatile private var initialized = false
   @volatile private var defaultRootLevel: Level = null
-  @volatile private var defaultSparkLog4jConfig = false
+  @volatile private var shouldUseDefaultSparkLog4jConfig = false
+  @volatile private var defaultSparkLog4jConfigApplied = false
```

Comment:
Note: this is not useful. We can just remove it.

Reply:
Sorry for the late reply.

Not exactly. I found
```
Logging.initializeLogging()
Logging.uninitialize()
Logging.initializeLogging() # this **may** not work properly again
```

To fix the semantic, it introduces a flag to memorize the action of the first call `initializeLogging`

Reply:
```
$ git grep defaultSparkLog4jConfigApplied
common/utils/src/main/scala/org/apache/spark/internal/Logging.scala:        Logging.defaultSparkLog4jConfigApplied = true
common/utils/src/main/scala/org/apache/spark/internal/Logging.scala:  @volatile private var defaultSparkLog4jConfigApplied = false
common/utils/src/main/scala/org/apache/spark/internal/Logging.scala:        defaultSparkLog4jConfigApplied = false
```

The variable `defaultSparkLog4jConfigApplied` is write-only and does not appear to be used in any conditional checks.

Reply:
hmm... seems one flag is sufficient, let me change and test

Reply:
tested, it works well with one flag too.